### PR TITLE
Expose some API so that the library can also be used with other systems such as React.

### PR DESCRIPTION
**Pull request by SY102134**
Allow use EmojiView UI alone,
Allow use EmojiManager.replaceWithImages function alone.

### CODE REVIEW

File: emoji/src/main/java/com/vanniktech/emoji/EmojiImageView.java [hunk 1]
```diff
@@ -15,7 +15,7 @@
 import com.vanniktech.emoji.listeners.OnEmojiClickListener;
 import com.vanniktech.emoji.listeners.OnEmojiLongClickListener;
 
-public final class EmojiImageView extends AppCompatImageView {
+public class EmojiImageView extends AppCompatImageView {
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiImageView.java]: Why do you need this change?

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPagerAdapter.java [hunk 2]
```diff
@@ -6,7 +6,7 @@
 import com.vanniktech.emoji.listeners.OnEmojiClickListener;
 import com.vanniktech.emoji.listeners.OnEmojiLongClickListener;
 
-final class EmojiPagerAdapter extends PagerAdapter {
+public class EmojiPagerAdapter extends PagerAdapter {
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiPagerAdapter.java]: why the need to remove `final`?

File: emoji/src/main/java/com/vanniktech/emoji/EmojiVariantPopup.java [hunk 3]
```diff
@@ -20,7 +20,7 @@
 
 import static android.view.View.MeasureSpec.makeMeasureSpec;
 
-final class EmojiVariantPopup {
+public class EmojiVariantPopup {
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiVariantPopup.java]: same

File: emoji/src/main/java/com/vanniktech/emoji/EmojiView.java [hunk 4]
```diff
@@ -23,7 +23,7 @@
 import com.vanniktech.emoji.listeners.RepeatListener;
 import java.util.concurrent.TimeUnit;
 
-@SuppressLint("ViewConstructor") final class EmojiView extends LinearLayout implements ViewPager.OnPageChangeListener {
+@SuppressLint("ViewConstructor") public class EmojiView extends LinearLayout implements ViewPager.OnPageChangeListener {
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiView.java]: same

File: emoji/src/main/java/com/vanniktech/emoji/EmojiView.java [hunk 5]
```diff
@@ -89,6 +92,10 @@ private void handleOnClicks(final ViewPager emojisPager) {
     }));
   }
 
+  public void setOsetOffscreenPageLimit(final ViewPager emojisPager, final EmojiPagerAdapter emojiPager) {
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiView.java]: I don't quite understand why you'd need this.

File: emoji/src/main/java/com/vanniktech/emoji/RecentEmojiManager.java [hunk 6]
```diff
@@ -12,7 +12,7 @@
 import java.util.List;
 import java.util.StringTokenizer;
 
-final class RecentEmojiManager implements RecentEmoji {
+public class RecentEmojiManager implements RecentEmoji {
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/RecentEmojiManager.java]: same

File: emoji/src/main/java/com/vanniktech/emoji/VariantEmojiManager.java [hunk 7]
```diff
@@ -8,7 +8,7 @@
 import java.util.List;
 import java.util.StringTokenizer;
 
-@SuppressWarnings("PMD.ForLoopCanBeForeach") final class VariantEmojiManager implements VariantEmoji {
+@SuppressWarnings("PMD.ForLoopCanBeForeach") public class VariantEmojiManager implements VariantEmoji {
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/VariantEmojiManager.java]: same

**Comment by SY102134** [comment]
@vanniktech 
Sorry, I am not an Android programmer, so I did some wrong things!
I will change `public class` to `public final class`, what do you think?
For `setOsetOffscreenPageLimit`, I have to set limit equal page count otherwise the emoji viewPager will not all show in React Native, So I need this function.

**Comment by SY102134** [comment]
Now, I modify `public class` to `public final class`, and remove `setOsetOffscreenPageLimit`.

@vanniktech , Do you have any good ways for me to set offscreenPageLimit equal  emoji viewPager count?

**Comment by SY102134** [comment]
Yes, Viewpager does not show when viewpager selected  on react native, the react native solution is this.[React native viewPager](https://github.com/facebook/react-native/blob/1e8f3b11027fe0a7514b4fc97d0798d3c64bc895/ReactAndroid/src/main/java/com/facebook/react/views/viewpager/ReactViewPager.java#L49)

**Comment by vanniktech** [comment]
So you need to switch out the normal ViewPager with a ReactViewPager?

**Comment by SY102134** [comment]
No, I need to set offscreenPageLimit equal emoji viewPager size keep them alive.

**Comment by vanniktech** [comment]
The `ReactViewPager` does a lot more than that.

**Comment by SY102134** [comment]
I clicked the wrong button to close this request.

**Comment by SY102134** [comment]
That's it if you do not have a good way to set offscreenPageLimit, i can rewrite the code on my local.

**Comment by SY102134** [comment]
@vanniktech , No need any changes now!

**Comment by vanniktech** [comment]
Thanks @SY102134  :)

**Comment by premacck** [comment]
Hey @vanniktech I just started using this library and found this very useful. Thank you very much!

One thing I have to ask though, I have implemented version 0.5.1 of this project through gradle dependency, and I still can't see the changes of this pull request. Are they yet to be released? I am very eager to explore this library.

**Comment by vanniktech** [comment]
For now, you need to use the 0.6.0-SNAPSHOT

**Pull request merged**